### PR TITLE
add getprice to estimate irys upload cost

### PR DESCRIPTION
**Pull request by jillxuu**
### Description
add getprice to estimate irys upload cost

### Related Links
https://docs.irys.xyz/developer-docs/irys-sdk/api/getPrice

### CODE REVIEW

File: src/plugins/assetUploader/index.ts [hunk 1]
```diff
@@ -80,4 +81,16 @@ export class AssetUploader {
...
+
+  /**
+   * Get cost estimate
+   *
+   * @param args.numBytes  The number of bytes to check the price for
+   *
+   * @return Cost to upload numBytes, unit is the token specified
+   * when instantiating the Irys object Return value is in atomic units
+   */
+  async getPrice(args: { numBytes: number }): Promise<BigNumber> {
```
[the remark this answers is not in the document]
Comment (jillxuu) [reply src/plugins/assetUploader/index.ts:93]: do they have `estimateFolderPrice` for node irys tho? didn't see sdk reference in the doc

File: src/plugins/assetUploader/irys.ts [hunk 2]
```diff
@@ -195,4 +196,24 @@ export class IrysAssetUploader implements IAssetUploader {
...
+   *
+   * @return Cost to upload numBytes, unit is the token specified
+   * when instantiating the Irys object Return value is in atomic units
+   */
+  async getPrice(args: { account: Account; numBytes: number }): Promise<BigNumber> {
+    const { account, numBytes } = args;
+    const irys = await this.getIrys({ account });
+
+    try {
+      return await irys.getPrice(numBytes);
```
Comment (0xmaayan) [inline src/plugins/assetUploader/irys.ts:214]: I would just return a `number` here and remove the use of `BigNumber`

```suggestion
      return await irys.getPrice(numBytes).toNumber;
```

File: src/plugins/assetUploader/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xmaayan) [reply src/plugins/assetUploader/index.ts:93]: I think utils is shared between web and node

File: package.json [hunk 3]
```diff
@@ -65,11 +65,13 @@
   "devDependencies": {
     "@aptos-labs/aptos-cli": "^0.1.2",
     "@babel/traverse": "^7.23.6",
+    "bignumber.js": "^9.1.2",
     "@graphql-codegen/cli": "^5.0.0",
     "@graphql-codegen/import-types-preset": "^3.0.0",
     "@graphql-codegen/typescript": "^4.0.1",
     "@graphql-codegen/typescript-graphql-request": "^6.0.1",
     "@graphql-codegen/typescript-operations": "^4.0.1",
+    "@irys/sdk": "0.1.15",
```
Comment (0xmaayan) [inline package.json:74]: Irys is already at `0.2.1` - might worth to update?

Comment (jillxuu) [reply package.json:74]: in the past ran into version mismatch issue. wanna get the functionality works first and then do version update in a separate PR.

**Review by 0xmaayan — APPROVED** [review]

**Comment by 0xaptosj** [comment]
not related to this PR, but when we calculate upload cost, how do we handle the size change after we update the media url in  metadata files? If the media url is blank in metadata file at the beginning, after the media upload and it's filled, the size of metadata file will grow a bit.

**Comment by 0xmaayan** [comment]
> not related to this PR, but when we calculate upload cost, how do we handle the size change after we update the media url in metadata files? If the media url is blank in metadata file at the beginning, after the media upload and it's filled, the size of metadata file will grow a bit.
> 
> Maybe a potential solution is we calculate how many media url there would be, e.g. for a collection of 1000 NFTs, there would be 1009+1 media urls need to be updated. And Irys URL should have fixed length, we can add the total file size by 1001 * bytes_of(Irys URL)

do you mean updating a folder after a folder has been uploaded to irys? or it the change in cost with vs without the media url?

**Comment by jillxuu** [comment]
> not related to this PR, but when we calculate upload cost, how do we handle the size change after we update the media url in metadata files? If the media url is blank in metadata file at the beginning, after the media upload and it's filled, the size of metadata file will grow a bit.
> 
> Maybe a potential solution is we calculate how many media url there would be, e.g. for a collection of 1000 NFTs, there would be 1009+1 media urls need to be updated. And Irys URL should have fixed length, we can add the total file size by 1001 * bytes_of(Irys URL)

this shouldn't be a problem from beginning because the cost should be calculated before the submission where the json file is already updated?

**Pull request merged**